pulsar-broker/src/test/java/org/apache/pulsar/client/api/KeySharedSubscriptionTest.java

```diff
@@ -295,6 +295,33 @@ public void testDisableKeySharedSubscription() throws PulsarClientException {
             .subscribe();
     }
 
+    @Test()
+    public void testCannotUseAcknowledgeCumulative() throws PulsarClientException {
+        this.conf.setSubscriptionKeySharedEnable(true);
+        String topic = "persistent://public/default/key_shared_ack_cumulative-" + UUID.randomUUID();
+
+        @Cleanup
+        Producer<Integer> producer = createProducer(topic, false);
+
+        @Cleanup
+        Consumer<Integer> consumer = createConsumer(topic);
+
+        for (int i = 0; i < 10; i++) {
+            producer.send(i);
+        }
+
+        for (int i = 0; i < 10; i++) {
+            Message<Integer> message = consumer.receive();
+            if (i == 9) {
+                try {
+                    consumer.acknowledgeCumulative(message);
```

Comment:
add `fail("should have failed")` after calling ack, to verify the call is indeed throwing exception